### DISCUSSION TITLE
v7.7.6

### DISCUSSION THREAD

**Discussion by davidmurdoch** [ConsenSys-archive/ganache#4284] [Releases]
<a id="user-content-v7.7.6-top"></a>
<h4>
  <p align="center">
		<code>&nbsp;<a href="#user-content-v7.7.6-fixes">Fixes</a>&nbsp;</code>
<img height="36" width="0" src="https://raw.githubusercontent.com/davidmurdoch/px/master/1px.gif">
		<code>&nbsp;<a href="#user-content-v7.7.6-changelog">Changelog</a>&nbsp;</code>
  <img height="36" width="0" src="https://raw.githubusercontent.com/davidmurdoch/px/master/1px.gif">
		<code>&nbsp;<a href="#user-content-v7.7.6-known-issues">Known&nbsp;Issues</a>&nbsp;</code>
  <img height="36" width="0" src="https://raw.githubusercontent.com/davidmurdoch/px/master/1px.gif">
		<code>&nbsp;<a href="#user-content-v7.7.6-future-plans">Future&nbsp;Plans</a>&nbsp;</code>
  <img height="36" width="0" src="https://raw.githubusercontent.com/davidmurdoch/px/master/1px.gif">
  </p>
</h4>

---

Thanks again to @MicaiahReid for his excellent work, and to @benams and @adjisb for their contributions!

If you have some time, we encourage you to browse our [issues](https://github.com/trufflesuite/ganache/issues) to find anything you'd like implemented/fixed sooner. Give them a +1 and we'll use this community feedback to help prioritize what we work on! Or better yet, [open a new issue](https://github.com/trufflesuite/ganache/issues/new), [open a PR](https://github.com/trufflesuite/ganache/compare) to fix an existing issue if you really want to get involved.

We've changed 5 files across 2 merged pull requests, tallying 174 additions and 112 deletions, since our last release.

<a id="user-content-v7.7.6-fixes" ></a>

---

# <p align="center"><a href="#user-content-v7.7.6-fixes"><img alt="Fixes" width="auto" src="https://raw.githubusercontent.com/trufflesuite/ganache/release-notes-assets/title-images/fixes.svg"></a></p>

- [fix: eth get block transaction count by hash to work with forks (#3739)](#user-content-v7.7.6-fixes-0)
- [fix: log warning regarding transactions with future-nonces when in eager mode (#4166)](#user-content-v7.7.6-fixes-1)

---

### <a id="user-content-v7.7.6-fixes-0"></a>fix: eth get block transaction count by hash to work with forks (#3739)

The RPC method `eth_getBlockTransactionCountByHash` had a bug that caused it to not return results when requesting a block before the effective "forkBlockNumber". 

<p align="right"><sup><a href="#user-content-v7.7.6-fixes">back to fixes</a></sup></p>

### <a id="user-content-v7.7.6-fixes-1"></a>fix: log warning regarding transactions with future-nonces when in eager mode (#4166)

@benams ran into a confusing situation in which future-nonce transactions that are sent to Ganache running in eager mining mode (the default) appear to hang indefinitely, rather than being mined or rejected. This is expected behavior in eager mode, as Ganache does not return the hash until _after_ it is mined. To help alleviate the confusion we now log a message about the status of these kinds of transactions.<br/>
Closes #4165 

<p align="right"><sup><a href="#user-content-v7.7.6-fixes">back to fixes</a></sup></p>

<p align="right"><sup><a href="#user-content-v7.7.6-top">back to top</a></sup></p>
        

<a id="user-content-v7.7.6-changelog" ></a>

---

# <p align="center"><a href="#user-content-v7.7.6-changelog"><img alt="Changelog" width="auto" src="https://raw.githubusercontent.com/trufflesuite/ganache/release-notes-assets/title-images/changelog.svg"></a></p>

 - #3739 fix: eth get block transaction count by hash to work with forks (@adjisb)
 - #4166 fix: log warning regarding transactions with future-nonces when in eager mode (@MicaiahReid)

<p align="right"><sup><a href="#user-content-v7.7.6-top">back to top</a></sup></p>
        

<a id="user-content-v7.7.6-known-issues" ></a>

---

# <p align="center"><a href="#user-content-v7.7.6-known-issues"><img alt="Known Issues" width="auto" src="https://raw.githubusercontent.com/trufflesuite/ganache/release-notes-assets/title-images/known-issues.svg"></a></p>

### Top Priority:
 - Add `eth_createAccessList` RPC method ([#1056](https://github.com/trufflesuite/ganache/issues/1056))
### Coming Soon™:
 - Implications failed:  fork.headers -> url ([#2627](https://github.com/trufflesuite/ganache/issues/2627))
 - In Geth chain-mode, logic to accept/reject transactions based on gas price/limit should match Geth ([#2176](https://github.com/trufflesuite/ganache/issues/2176))
 - `evm_mine` and `miner_start` don't respect --mode.instamine=eager ([#2029](https://github.com/trufflesuite/ganache/issues/2029))
 - `evm_setAccount*` is race-conditiony ([#1646](https://github.com/trufflesuite/ganache/issues/1646))
 - `@ganache/filecoin@alpha` doesn't work with `ganache@alpha` ([#1150](https://github.com/trufflesuite/ganache/issues/1150))
 - Launching ganache with fork is throwing revert errors when communicating with 3rd party contracts ([#956](https://github.com/trufflesuite/ganache/issues/956))
 - Build a real pending block! ([#772](https://github.com/trufflesuite/ganache/issues/772))
 - VM Exception when interfacing with Kyber contract ([#606](https://github.com/trufflesuite/ganache/issues/606))
 - After calling `evm_mine`, `eth_getLogs` returns same logs for all blocks ([#533](https://github.com/trufflesuite/ganache/issues/533))
 - personal_unlockAccount works with any password ([#165](https://github.com/trufflesuite/ganache/issues/165))
 - --db Option Requires Same Mnemonic and Network ID ([#1030](https://github.com/trufflesuite/ganache/issues/1030))

<p align="right"><sup><a href="#user-content-v7.7.6-top">back to top</a></sup></p>
        

<a id="user-content-v7.7.6-future-plans" ></a>

---

# <p align="center"><a href="#user-content-v7.7.6-future-plans"><img alt="Future Plans" width="auto" src="https://raw.githubusercontent.com/trufflesuite/ganache/release-notes-assets/title-images/future-plans.svg"></a></p>

### Top Priority:
 - Accept a genesis.json file ([#1042](https://github.com/trufflesuite/ganache/issues/1042))
### Coming Soon™:
 - Switch to esbuild to make build times faster/reasonable ([#1555](https://github.com/trufflesuite/ganache/issues/1555))
 - fork specific block & specific index ([#952](https://github.com/trufflesuite/ganache/issues/952))
 - Allow to sync forked chain to the latest block ([#643](https://github.com/trufflesuite/ganache/issues/643))
 - Implement a streaming trace capability ([#381](https://github.com/trufflesuite/ganache/issues/381))
 - Improve log performance when forking ([#145](https://github.com/trufflesuite/ganache/issues/145))
 - Log contract events ([#45](https://github.com/trufflesuite/ganache/issues/45))

<p align="right"><sup><a href="#user-content-v7.7.6-top">back to top</a></sup></p>
        

[Open new issues](https://github.com/trufflesuite/ganache/issues/new?milestone=7.0.0) to influence what we gets implemented and prioritized.

---

<p align="center">
  💖 The Truffle Team
</p>

<hr /><em>This discussion was created from the release <a href='https://github.com/trufflesuite/ganache/releases/tag/v7.7.6'>v7.7.6</a>.</em>

**Comment by MicaiahReid** [comment]
One more release until v7.7.7!
